flink-table/flink-sql-parser-hive/src/main/codegen/data/Parser.tdd

```diff
@@ -129,6 +130,7 @@
     "LINES"
     "LOAD"
     "LOCATION"
+    "MODULES"
```

Comment:
We should also add `MODULES` into non-reserved-keywords, otherwise it will break user jobs which uses `modules` as column name. For example, `select a, b, modules from T` will parse failed.

Reply:
Hi Jark, this is the point I've struggled with. The `USE database` syntax takes a compound identifier for the database name, and the non-reserved keyword will be matched and cause a parse failure.

The grammar for `USE db`
```
SqlUseDatabase SqlUseDatabase() :
{
    SqlIdentifier databaseName;
    SqlParserPos pos;
}
{
    <USE> { pos = getPos();}
    databaseName = CompoundIdentifier()
    {
        return new SqlUseDatabase(pos, databaseName);
    }
}
```

The grammar for `IdentiiferSegment` of `CompoundIdentifier`

```
void IdentifierSegment(List<String> names, List<SqlParserPos> positions) :
{
    final String id;
    char unicodeEscapeChar = BACKSLASH;
    final SqlParserPos pos;
    final Span span;
}
{
    (
        <IDENTIFIER> {
           ...
        }
    |
        <HYPHENATED_IDENTIFIER> {
            ...
        }
    |
        <QUOTED_IDENTIFIER> {
            ...
        }
    |
        <BACK_QUOTED_IDENTIFIER> {
           ...
        }
    |
        <BRACKET_QUOTED_IDENTIFIER> {
           ...
        }
    |
        <UNICODE_QUOTED_IDENTIFIER> {
            ...
        }
        [
            <UESCAPE> <QUOTED_STRING> {
                ...
            }
        ]
        {
            ...
        }
    |
        id = NonReservedKeyWord() {
            pos = getPos();
        }
    )
  ...
}
```
You can check the parser.jj template or generated parser.jj file to identify it, and I'm open to your suggestions

Reply:
Thanks. Then we should warn users this is a reserved keyword now. I will add this information into release note once PR is merged. 

`The term MODULES is a reserved keyword now. Use backticks to escape column names and other identifiers with this name.`